mirai-core-mock/src/internal/contact/MockNormalMemberImpl.kt

```diff
@@ -0,0 +1,237 @@
+            throw PermissionDeniedException("Bot is not the owner of group ${group.id}, can't modify the permission of $id($permission")
+        }
+        if (operation && permission > MemberPermission.MEMBER) return
+
+        if (permission == MemberPermission.OWNER) {
+            throw IllegalArgumentException("Not allowed modify permission of owner ($id, $permission)")
+        }
+        val newPerm = if (operation) MemberPermission.ADMINISTRATOR else MemberPermission.MEMBER
+        if (newPerm != permission) {
+            val oldPerm = permission
+            mockApi.permission = oldPerm
+            MemberPermissionChangeEvent(this, oldPerm, newPerm).broadcast()
+        }
+    }
+
+    override suspend fun sendMessage(message: Message): MessageReceipt<NormalMember> {
+        return super<AbstractMockContact>.sendMessage(message).cast()
+    }
+
+    override suspend fun mute(durationSeconds: Int) {
+        requireBotPermissionHigherThanThis("mute")
+        require(durationSeconds > 0) {
+            "$durationSeconds < 0"
+        }
+        mockApi.muteTimeEndTimestamp = currentTimeSeconds() + durationSeconds
+        MemberMuteEvent(this, durationSeconds, null)
+    }
+
+    override suspend fun broadcastMute(target: MockNormalMember, durationSeconds: Int) {
+        target.mockApi.muteTimeEndTimestamp = currentTimeSeconds() + durationSeconds
+        if (target.id == bot.id) {
+            if (durationSeconds == 0) {
+                BotUnmuteEvent(this)
+            } else {
+                BotMuteEvent(durationSeconds, this)
+            }
+        } else {
+            if (durationSeconds == 0) {
+                MemberUnmuteEvent(target, this)
+            } else {
+                MemberMuteEvent(target, durationSeconds, this)
+            }
+        }.broadcast()
+    }
+
+    override suspend fun says(message: MessageChain): MessageChain {
+        val src = newMsgSrc(true, message) { ids, internalIds, time ->
+            mockApi.lastSpeakTimestamp = time
+            OnlineMsgSrcFromGroup(ids, internalIds, time, message, bot, this)
+        }
+        val msg = src.withMessage(message)
+        GroupMessageEvent(nameCardOrNick, permission, this, msg, src.time).broadcast()
+        return msg
+    }
+
+    override fun newMessagePreSend(message: Message): MessagePreSendEvent {
+        return GroupTempMessagePreSendEvent(this, message)
+    }
+
+    override suspend fun postMessagePreSend(message: MessageChain, receipt: MessageReceipt<*>) {
+        GroupTempMessagePostSendEvent(this, message, null, receipt.cast()).broadcast()
+    }
+
+    override fun newMessageSource(message: MessageChain): OnlineMessageSource.Outgoing {
+        return newMsgSrc(false, message) { ids, internalIds, time ->
+            OnlineMsgSrcToTemp(ids, internalIds, time, message, bot, bot, this)
+        }
+    }
+
+    override fun toString(): String {
+        return "$nameCardOrNick[$id]"
```

Comment:
```suggestion
        return "NormalMember(name=$nameCardOrNick, id=$id)"
```

你的各种 contact 类型的 toString 一共有三种格式哦